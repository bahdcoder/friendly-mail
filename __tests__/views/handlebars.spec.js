const HandlebarsRenderEngine = require('../../src/Views/Handlebars')

describe('The HandlebarsRenderEngine', () => {
  it('it instantiates properly', () => {
    const config = {
      viewEngine: 'handlebars'
    }

    expect(new HandlebarsRenderEngine(config)).toMatchSnapshot()
  })

  it('can render the emails content for all three email types', () => {
    const config = {
      viewEngine: 'handlebars',
      views: '__tests__/__mocks__/mails'
    }

    expect(
      new HandlebarsRenderEngine(config).render('confirm-email', {
        username: 'bahdcoder'
      })
    ).toMatchSnapshot()
  })
})
