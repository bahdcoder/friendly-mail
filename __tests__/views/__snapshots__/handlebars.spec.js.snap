// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The HandlebarsRenderEngine can render the emails content for all three email types 1`] = `
Object {
  "html": "<strong>
  HEY bahdcoder
</strong>
<br>
CONFIRM YOUR EMAIL
",
  "text": "hey bahdcoder, please confirm your email.
",
  "watchHtml": "hey bahdcoder, please confirm your email.
",
}
`;

exports[`The HandlebarsRenderEngine it instantiates properly 1`] = `
HandleBarsRenderEngine {
  "Config": Object {
    "viewEngine": "handlebars",
  },
  "enginesExtensionsMap": Object {
    "edge": "edge",
    "handlebars": "hbs",
  },
}
`;
